src/coord/src/sink_connector.rs

```diff
@@ -118,34 +118,66 @@ async fn get_latest_ts(
     // Seek to end-1 offset
     let mut tps = TopicPartitionList::new();
     tps.add_partition(consistency_topic, partition);
-    tps.set_partition_offset(consistency_topic, partition, Offset::OffsetTail(1))?;
+    tps.set_partition_offset(consistency_topic, partition, Offset::Beginning)?;
 
-    consumer
-        .assign(&tps)
-        .with_context(|| format!("Error seeking in consistency topic {}", consistency_topic))?;
+    consumer.assign(&tps).with_context(|| {
+        format!(
+            "Error seeking in consistency topic {}:{}",
+            consistency_topic, partition
+        )
+    })?;
+
+    // We scan from the beginning and see if we can find an END record. We have
+    // to do it like this because Kafka Control Batches mess with offsets. We
+    // thereforecannot simply take the last offset from the back and expect an
```

Comment:
nit: need to have a space between therefore and cannot

Reply:
fixing